org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/AbstractJavaProjectConfigurator.java

```diff
@@ -496,10 +510,14 @@ private void addResourceDirs(IClasspathDescriptor classpath, IProject project, M
       if(directory == null) {
         continue;
       }
-      File resourceDirectory = new File(directory);
-      IPath relativePath = getProjectRelativePath(project, directory);
-      IResource r = project.findMember(relativePath);
-      if(r == project) {
+      File resourceDirectory = new File(resource.getDirectory());
+      String resourceCanonicalPath = resourceDirectory.getAbsolutePath();
+      try {
+        resourceCanonicalPath = resourceDirectory.getCanonicalPath();
+      } catch(IOException ex) {
+      }
```

Comment:
I have not yet tried this out, but I'm not sure this is correct.
If `resource.getDirectory()` is a relative path (haven't checked if that can happen), then it will resolve against the running directory of the application, which will be something like the IDE location.
But here it should be resolved against the project directory, shouldn't it?

Furthermore if `resourceDirectory.getAbsolutePath()` is the fall back in case `getCanonicalPath()` throws, it should only be called in the catch block.

Reply:
When i checked resource.getDirectory() were only absolutePath no relativePath